### ISSUE TITLE
Crash on resize since 0.3

### ISSUE THREAD

**Issue by tarkah** [hecrj/iced#816]
Since updating [Ajour](https://github.com/ajour/ajour) to iced 0.3, I'm crashing when trying to resize the program.

Below is the panic. Is this a known upstream issue? We've confirmed with other users that they experience this as well. Interestingly, it only appears to crash if we set `Application::scale_factor` to <>1.0.

```
wgpu error: Validation Error

Caused by:
    In a RenderPass
      note: encoder = `iced_wgpu encoder`
    In a set_scissor_rect command
    Invalid ScissorRect parameters


15:17:00.501 [panic][ERROR] thread 'main' panicked at 'Handling wgpu errors as fatal by default': C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\wgpu-0.7.1\src\backend\direct.rs:1896
   0: backtrace::backtrace::trace_unsynchronized<closure-0>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\backtrace-0.3.56\src\backtrace\mod.rs:66
   1: backtrace::backtrace::trace<closure-0>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\backtrace-0.3.56\src\backtrace\mod.rs:53
   2: backtrace::capture::Backtrace::create
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\backtrace-0.3.56\src\capture.rs:176
   3: backtrace::capture::Backtrace::new
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\backtrace-0.3.56\src\capture.rs:140
   4: log_panics::init::{{closure}}
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\log-panics-2.0.0\src\lib.rs:52
   5: std::panicking::rust_panic_with_hook
             at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca\/library\std\src\panicking.rs:597
   6: std::panicking::begin_panic::{{closure}}<str*>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs:522
   7: std::sys_common::backtrace::__rust_end_short_backtrace<closure-0,!>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\sys_common\backtrace.rs:141
   8: std::panicking::begin_panic<str*>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs:521
   9: wgpu::backend::direct::default_error_handler
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\wgpu-0.7.1\src\backend\direct.rs:1896
  10: core::ops::function::Fn::call<fn(wgpu::Error),tuple<wgpu::Error>>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\ops\function.rs:70
  11: alloc::boxed::{{impl}}::call<tuple<wgpu::Error>,UncapturedErrorHandler,alloc::alloc::Global>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\alloc\src\boxed.rs:1321
  12: wgpu::backend::direct::ErrorSinkRaw::handle_error
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\wgpu-0.7.1\src\backend\direct.rs:1883
  13: wgpu::backend::direct::Context::handle_error<wgpu_core::command::render::RenderPassError>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\wgpu-0.7.1\src\backend\direct.rs:109
  14: wgpu::backend::direct::{{impl}}::command_encoder_end_render_pass
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\wgpu-0.7.1\src\backend\direct.rs:1739
  15: wgpu::{{impl}}::drop
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\wgpu-0.7.1\src\lib.rs:2486
  16: core::ptr::drop_in_place<wgpu::RenderPass>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\ptr\mod.rs:175
  17: iced_wgpu::quad::Pipeline::draw
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\iced_wgpu-0.4.0\src\quad.rs:275
  18: iced_wgpu::backend::Backend::flush
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\iced_wgpu-0.4.0\src\backend.rs:116
  19: iced_wgpu::backend::Backend::draw<alloc::string::String>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\iced_wgpu-0.4.0\src\backend.rs:82
  20: iced_wgpu::window::compositor::{{impl}}::draw<alloc::string::String>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\iced_wgpu-0.4.0\src\window\compositor.rs:156
  21: iced_winit::application::run_instance::{{closure}}<iced::application::Instance<ajour::gui::Ajour>,iced::executor::platform::Default,iced_wgpu::window::compositor::Compositor>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\iced_winit-0.3.0\src\application.rs:350
  22: core::future::from_generator::{{impl}}::poll<generator-0>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\future\mod.rs:80
  23: iced_winit::application::run::{{closure}}<iced::application::Instance<ajour::gui::Ajour>,iced::executor::platform::Default,iced_wgpu::window::compositor::Compositor>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\iced_winit-0.3.0\src\application.rs:193
  24: winit::platform_impl::platform::event_loop::{{impl}}::run_return::{{closure}}<ajour::gui::Message,closure-1>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop.rs:203
  25: alloc::boxed::{{impl}}::call_mut<tuple<winit::event::Event<ajour::gui::Message>, mut winit::event_loop::ControlFlow*>,FnMut<tuple<winit::event::Event<ajour::gui::Message>, mut winit::event_loop::ControlFlow*>>,alloc::alloc::Global>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\alloc\src\boxed.rs:1314
  26: winit::platform_impl::platform::event_loop::runner::{{impl}}::call_event_handler::{{closure}}<ajour::gui::Message>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop\runner.rs:245
  27: std::panic::{{impl}}::call_once<tuple<>,closure-0>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panic.rs:322
  28: std::panicking::try::do_call<std::panic::AssertUnwindSafe<closure-0>,tuple<>>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs:381
  29: std::panicking::try::do_catch<std::panic::AssertUnwindSafe<closure-0>,alloc::collections::linked_list::LinkedList<alloc::vec::Vec<ajour_core::addon::AddonFolder>>> 
  30: std::panicking::try<tuple<>,std::panic::AssertUnwindSafe<closure-0>>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs:345
  31: std::panic::catch_unwind<std::panic::AssertUnwindSafe<closure-0>,tuple<>>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panic.rs:396
  32: winit::platform_impl::platform::event_loop::runner::EventLoopRunner<ajour::gui::Message>::catch_unwind<ajour::gui::Message,tuple<>,closure-0>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop\runner.rs:152
  33: winit::platform_impl::platform::event_loop::runner::EventLoopRunner<ajour::gui::Message>::call_event_handler<ajour::gui::Message>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop\runner.rs:239
  34: winit::platform_impl::platform::event_loop::runner::EventLoopRunner<ajour::gui::Message>::send_event<ajour::gui::Message>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop\runner.rs:210
  35: winit::platform_impl::platform::event_loop::SubclassInput<ajour::gui::Message>::send_event<ajour::gui::Message>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop.rs:93
  36: winit::platform_impl::platform::event_loop::public_window_callback::{{closure}}<ajour::gui::Message>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop.rs:835
  37: core::ops::function::FnOnce::call_once<closure-0,tuple<>>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\ops\function.rs:227
  38: std::panic::{{impl}}::call_once<isize,closure-0>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panic.rs:322
  39: std::panicking::try::do_call<std::panic::AssertUnwindSafe<closure-0>,isize>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs:381
  40: std::panicking::try::do_catch<std::panic::AssertUnwindSafe<closure-0>,alloc::collections::linked_list::LinkedList<alloc::vec::Vec<ajour_core::addon::AddonFolder>>> 
  41: std::panicking::try<isize,std::panic::AssertUnwindSafe<closure-0>>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs:345
  42: std::panic::catch_unwind<std::panic::AssertUnwindSafe<closure-0>,isize>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panic.rs:396
  43: winit::platform_impl::platform::event_loop::runner::EventLoopRunner<ajour::gui::Message>::catch_unwind<ajour::gui::Message,isize,closure-0>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop\runner.rs:152
  44: winit::platform_impl::platform::event_loop::public_window_callback<ajour::gui::Message>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop.rs:1912
  45: DefSubclassProc
  46: DefSubclassProc
  47: CallWindowProcW
  48: CallWindowProcW
  49: glPushClientAttrib
  50: CallWindowProcW
  51: DispatchMessageW
  52: SendMessageTimeoutW
  53: KiUserCallbackDispatcher
  54: NtUserMessageCall
  55: GetWindowTextW
  56: IsIconic
  57: GetUserColorPreference
  58: Ordinal96
  59: IsCompositionActive
  60: IsCompositionActive
  61: GetWindowTextW
  62: winit::platform_impl::platform::event_loop::public_window_callback::{{closure}}<ajour::gui::Message>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop.rs:990
  63: core::ops::function::FnOnce::call_once<closure-0,tuple<>>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\ops\function.rs:227
  64: std::panic::{{impl}}::call_once<isize,closure-0>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panic.rs:322
  65: std::panicking::try::do_call<std::panic::AssertUnwindSafe<closure-0>,isize>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs:381
  66: std::panicking::try::do_catch<std::panic::AssertUnwindSafe<closure-0>,alloc::collections::linked_list::LinkedList<alloc::vec::Vec<ajour_core::addon::AddonFolder>>> 
  67: std::panicking::try<isize,std::panic::AssertUnwindSafe<closure-0>>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs:345
  68: std::panic::catch_unwind<std::panic::AssertUnwindSafe<closure-0>,isize>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panic.rs:396
  69: winit::platform_impl::platform::event_loop::runner::EventLoopRunner<ajour::gui::Message>::catch_unwind<ajour::gui::Message,isize,closure-0>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop\runner.rs:152
  70: winit::platform_impl::platform::event_loop::public_window_callback<ajour::gui::Message>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop.rs:1912
  71: DefSubclassProc
  72: DefSubclassProc
  73: CallWindowProcW
  74: CallWindowProcW
  75: glPushClientAttrib
  76: CallWindowProcW
  77: DispatchMessageW
  78: SendMessageTimeoutW
  79: KiUserCallbackDispatcher
  80: NtUserMessageCall
  81: GetWindowTextW
  82: IsIconic
  83: GetUserColorPreference
  84: Ordinal96
  85: IsCompositionActive
  86: IsCompositionActive
  87: GetWindowTextW
  88: CallWindowProcW
  89: CallWindowProcW
  90: DefSubclassProc
  91: DefSubclassProc
  92: winit::platform_impl::platform::event_loop::public_window_callback::{{closure}}<ajour::gui::Message>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop.rs:793
  93: core::ops::function::FnOnce::call_once<closure-0,tuple<>>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\ops\function.rs:227
  94: std::panic::{{impl}}::call_once<isize,closure-0>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panic.rs:322
  95: std::panicking::try::do_call<std::panic::AssertUnwindSafe<closure-0>,isize>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs:381
  96: std::panicking::try::do_catch<std::panic::AssertUnwindSafe<closure-0>,alloc::collections::linked_list::LinkedList<alloc::vec::Vec<ajour_core::addon::AddonFolder>>> 
  97: std::panicking::try<isize,std::panic::AssertUnwindSafe<closure-0>>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panicking.rs:345
  98: std::panic::catch_unwind<std::panic::AssertUnwindSafe<closure-0>,isize>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\panic.rs:396
  99: winit::platform_impl::platform::event_loop::runner::EventLoopRunner<ajour::gui::Message>::catch_unwind<ajour::gui::Message,isize,closure-0>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop\runner.rs:152
 100: winit::platform_impl::platform::event_loop::public_window_callback<ajour::gui::Message>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop.rs:1912
 101: DefSubclassProc
 102: DefSubclassProc
 103: CallWindowProcW
 104: CallWindowProcW
 105: glPushClientAttrib
 106: CallWindowProcW
 107: DispatchMessageW
 108: winit::platform_impl::platform::event_loop::EventLoop<ajour::gui::Message>::run_return<ajour::gui::Message,closure-1>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop.rs:218
 109: winit::platform_impl::platform::event_loop::EventLoop<ajour::gui::Message>::run<ajour::gui::Message,closure-1>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\platform_impl\windows\event_loop.rs:188
 110: winit::event_loop::EventLoop<ajour::gui::Message>::run<ajour::gui::Message,closure-1>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\winit-0.24.0\src\event_loop.rs:154
 111: iced_winit::application::run<iced::application::Instance<ajour::gui::Ajour>,iced::executor::platform::Default,iced_wgpu::window::compositor::Compositor>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\iced_winit-0.3.0\src\application.rs:168
 112: iced::application::Application::run<ajour::gui::Ajour>
             at C:\Users\cforsstrom\.cargo\registry\src\github.com-1ecc6299db9ec823\iced-0.3.0\src\application.rs:219
 113: ajour::gui::run
             at src\gui\mod.rs:1156
 114: ajour::main
             at src\main.rs:100
 115: core::ops::function::FnOnce::call_once<fn(),tuple<>>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\core\src\ops\function.rs:227
 116: std::sys_common::backtrace::__rust_begin_short_backtrace<fn(),tuple<>>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\sys_common\backtrace.rs:125
 117: std::rt::lang_start::{{closure}}<tuple<>>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\rt.rs:66
 118: core::ops::function::impls::{{impl}}::call_once
             at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca\library\core\src\ops\function.rs:280
 119: std::panicking::try::do_call
             at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca\/library\std\src\panicking.rs:381
 120: std::panicking::try
             at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca\/library\std\src\panicking.rs:345
 121: std::panic::catch_unwind
             at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca\/library\std\src\panic.rs:396
 122: std::rt::lang_start_internal
             at /rustc/e1884a8e3c3e813aada8254edfa120e85bf5ffca\/library\std\src\rt.rs:51
 123: std::rt::lang_start<tuple<>>
             at C:\Users\cforsstrom\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\rt.rs:65
 124: main
 125: invoke_main
             at d:\agent\_work\63\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:78
 126: __scrt_common_main_seh
             at d:\agent\_work\63\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288
 127: BaseThreadInitThunk
 128: RtlUserThreadStart
```

**Comment by 13r0ck** [comment]
I cloned the [Ajour development branch](https://github.com/ajour/ajour/tree/development) and I don't get the crashing issue. I'm on Linux, Is everyone who is having this issue is on Windows?

**Comment by tarkah** [comment]
We can reproduce on all platforms. Go to settings and change your scale up or down then try resizing. It only occurs when scale isn't 1.0

**Comment by hecrj** [comment]
Given that the issue seems to be absent when the `scale_factor` is `1.0`, I believe it may be related to a floating point precision error.

I will investigate further soon.

**Comment by zyansheep** [comment]
I have this issue as well (Linux/NixOS) in many of the example programs (and my own custom ones). Even when I implement the scale_factor function and return 1 for my application, it still crashes with the same error.

**Comment by hecrj** [comment]
Fixed by #818.
